File: problem/Binary_Representation_One/numSteps.js
/**
 * @param {string} s
 * @return {number}
 */
const numSteps = s => {
    let steps = 0, carry = 0;
    for (let i = s.length - 1; i > 0; i--) {
        const bit = s.charCodeAt(i) & 1;
        steps += 1 + (bit ^ carry);
        carry |= bit;
    }

    return steps + carry;
};

File: problem/Binary_Representation_One/numSteps.c
int numSteps(char *s)
{
    int steps = 0, carry = 0;
    for (int i = strlen(s) - 1; i > 0; i--)
    {
        int bit = s[i] & 1;
        steps += 1 + (bit ^ carry);
        carry |= bit;
    }
    return steps + carry;
}